src/analysis/fourier/riemann_lebesgue_lemma.lean

```diff
@@ -0,0 +1,152 @@
+    rw [complex.of_real_add, mul_add, add_comm, exp_add, ←neg_mul],
+    replace ht := complex.of_real_ne_zero.mpr ht,
+    have : I * ↑t * ↑(π / t) = π * I, by { field_simp, ring},
+    rw [this, exp_pi_mul_I], ring_nf, },
+  rw [this, integral_add_right_eq_self],
+  simp_rw [neg_smul, integral_neg],
+end
+
+/-- Rewrite the Riemann-Lebesgue integral in a form that allows us to use uniform continuity. -/
+lemma fourier_integral_eq_half_sub_half_period_translate
+  {t : ℝ} (ht : t ≠ 0) (hf : integrable f) :
+  ∫ x:ℝ, exp (I * t * x) • f x = (1 / (2 : ℂ)) • ∫ x:ℝ, exp (I * t * x) • (f x - f (x + π / t)) :=
+begin
+  simp_rw [smul_sub],
+  rw [integral_sub, fourier_integral_half_period_translate ht, sub_eq_add_neg, neg_neg,
+    ←two_smul ℂ _, ←@smul_assoc _ _ _ _ _ _ (is_scalar_tower.left ℂ), smul_eq_mul],
+  norm_num,
+  exacts [fourier_integrand_integrable hf t,
+    fourier_integrand_integrable (hf.comp_add_right (π / t)) t],
+end
+
+/-- Riemann-Lebesgue Lemma for continuous and compactly-supported functions: the integral
+`∫ x, exp (I * t * x) • f x` tends to 0 as `t` gets large.  -/
+lemma tendsto_integral_mul_exp_at_top_of_continuous_compact_support
+  (hf1 : continuous f) (hf2 : has_compact_support f) :
+  tendsto (λ t:ℝ, ∫ x:ℝ, exp (I * t * x) • f x) at_top (𝓝 0) :=
+begin
+  simp_rw [normed_add_comm_group.tendsto_nhds_zero, eventually_at_top, ge_iff_le],
+  intros ε hε,
+  -- Extract an explicit candidate bound on `t` from uniform continuity.
+  obtain ⟨R, hR1, hR2⟩ := hf2.exists_pos_le_norm,
+  obtain ⟨δ, hδ1, hδ2⟩ := metric.uniform_continuous_iff.mp
+    (hf2.uniform_continuous_of_continuous hf1) (ε / (1 + 2 * R)) (div_pos hε (by positivity)),
+  refine ⟨max π (1 + π / δ), λ t ht, _⟩,
+  have tpos : 0 < t := lt_of_lt_of_le real.pi_pos ((le_max_left _ _).trans ht),
+  -- Rewrite integral in terms of `f x - f (x + π / t)`.
+  rw fourier_integral_eq_half_sub_half_period_translate (lt_of_lt_of_le
+    (lt_max_of_lt_left real.pi_pos) ht).ne' (hf1.integrable_of_has_compact_support hf2),
+  rw [norm_smul, norm_eq_abs, ←complex.of_real_one, ←of_real_bit0, ←of_real_div,
+    complex.abs_of_nonneg one_half_pos.le],
+  have : ε = (1 / 2) * (2 * ε), by { field_simp, ring, },
+  rw [this, mul_lt_mul_left (one_half_pos : (0:ℝ) < 1/2)],
+  have : ‖∫ (x : ℝ), exp (I * ↑t * ↑x) • (f x - f (x + π / t))‖ ≤ ∫ (x : ℝ),
+    ‖exp (I * ↑t * ↑x) • (f x - f (x + π / t))‖, from norm_integral_le_integral_norm _,
+  refine lt_of_le_of_lt this _,
+  have : ∀ (x : ℝ), ‖exp (I * t * x)‖ = 1,
+  { intro x, rw [mul_assoc, ←complex.of_real_mul, mul_comm, norm_eq_abs, abs_exp_of_real_mul_I], },
+  simp_rw [norm_smul, this, one_mul],
+  -- Show integral can be taken over `[-(R + 1), R] ⊂ ℝ`.
+  let A := Icc (-(R + 1)) R,
+  have int_Icc :
+    ∫ (x : ℝ), ‖f x - f (x + π / t)‖ = ∫ x in A, ‖f x - f (x + π / t)‖,
+  { rw ←integral_indicator (measurable_set_Icc : measurable_set A),
+    congr' 1 with x,
+    symmetry,
+    refine (indicator_apply_eq_self.mpr (λ hx, _)),
```

Comment:
This could probably benefit from #18395, but since your PR will very likely be merged before I'll make the fix in #18395.

Reply:
Looks like #18395 is on its way to being merged, so we can fix any fallout here.

Reply:
Yes, it has been merged. Can you merge master, and see if it helps a little bit here?

Reply:
I merged the updated master, but I don't think there are actually any changes to the lemmas used here (both `set_integral_nonneg` and `integral_indicator` seem to take a measurability argument).